src/main/java/controller/ListMessages.java

```diff
@@ -199,15 +204,31 @@ public void listMessagesWithoutChatIdPathVariable(HttpServletRequest request) {
          * If it is null, sets it to the time when listMessages was called.
          */
         List<Message> messageListForReceivedTsUpdate = new ArrayList<Message>();
+        List<Long> attachmentIdList = new ArrayList<Long>();
```

Comment:
Do we have ImmutableList.Builder if yes, prefer to use that.

Reply:
+1

Reply:
Done

Reply:
Not done.
Here also, you can do the following instead.

```
ImmutableList.Builder<Long> attachmentIdListBuilder = ImmutableList.builder();
...
message.getAttachmentId().ifPresent(attachmentIdListBuilder::add);
...
ImmutableList<Long> attachmentIdList = attachmentIdListBuilder.build();
```

Reply:
Done